core/src/main/java/org/mule/DefaultMuleEvent.java

```diff
@@ -640,7 +640,7 @@ public String getMessageAsString(String encoding) throws MuleException
     {
         try
         {
-            return message.getPayloadForLogging(encoding);
+            return getMuleContext().getTransformationService().getPayloadAsString(message, encoding);
```

Comment:
Because this looks like an obvious bug/inconsistency, especially given this method throws an exception.